test/benchmarks/benchmark_slidehash.cc

```diff
@@ -0,0 +1,78 @@
+#include <stdint.h>
+#include <stdint.h>
+#include <limits.h>
+
+#include "zbuild.h"
+
+extern "C" {
+#include "deflate.h"
+}
+
+#include "cpu_features.h"
+
+#include <benchmark/benchmark.h>
+
+#define MAX_RANDOM_INTS 32768
+
+class slide_hash: public benchmark::Fixture {
+public:
+    uint16_t *l0;
+    uint16_t *l1;
+    deflate_state *s_g;
+    void SetUp(const ::benchmark::State& state) {
+        int32_t random_ints_size = MAX_RANDOM_INTS * sizeof(uint16_t);
+        l0 = (uint16_t *)malloc(HASH_SIZE * sizeof(uint16_t));
+        l1 = (uint16_t *)malloc(random_ints_size);
+
+        for (int32_t i = 0; i < MAX_RANDOM_INTS; i++) {
+            l1[i] = rand();
+        }
+
+        for (int32_t i = 0; i < HASH_SIZE; i++) {
+            l0[i] = rand();
+        }
+
+        deflate_state *s = (deflate_state*)malloc(sizeof(deflate_state));
+        s->head = l0;
+        s->prev = l1;
+        s_g = s;
+    }
+
+    void TearDown(const ::benchmark::State& state) {
+        free(l0);
+        free(l1);
+    }
+};
+
+BENCHMARK_DEFINE_F(slide_hash, c)(benchmark::State& state) {
+    s_g->w_size = (uint32_t)state.range(0);
+
+    for (auto _ : state) {
+        slide_hash_c(s_g);
+        benchmark::DoNotOptimize(s_g);
+    }
+}
+
+BENCHMARK_REGISTER_F(slide_hash, c)->DenseRange(2048, MAX_RANDOM_INTS, 2048);
+#ifdef X86_SSSE3_ADLER32
+BENCHMARK_DEFINE_F(slide_hash, sse2)(benchmark::State& state) {
+    s_g->w_size = (uint32_t)state.range(0);
+
+    for (auto _ : state) {
+        slide_hash_sse2(s_g);
+        benchmark::DoNotOptimize(s_g);
+    }
+}
+BENCHMARK_REGISTER_F(slide_hash, sse2)->DenseRange(2048, MAX_RANDOM_INTS, 2048);
```

Comment:
You'll want to modify this to use RangeMultiplier(2)->Range() like in my latest gist - that way we only test the possible sizes for slide_hash.

Reply:
Done.